oeps/architectural-decisions/oep-0065-frontend-composability.rst

```diff
@@ -0,0 +1,492 @@
+
+Reusable components (Paragon)
+-----------------------------
+
+MFEs may use different versions of ``@openedx/paragon``, resulting in functional and stylistic differences.
+
+Header/Footer
+-------------
+
+MFEs may use different versions of ``@edx/frontend-component-header`` and ``@edx/frontend-component-footer``, also resulting in functional, stylistic, and content/navigation differences.
+
+Branding
+--------
+
+Brand packages created from ``@openedx/brand-openedx`` may be different versions, resulting in any number of subtle visual differences.
+
+Other Dependencies
+------------------
+
+MFEs may have completely different versions of any other dependency. We mitigate some of this by consolidating some important dependencies in ``@openedx/frontend-build`` and ``@edx/frontend-platform``, but even those can have different versions from MFE to MFE. For developers, this increases cognitive load and slows velocity because of the need to adjust to the idiosyncracies of each application.
+
+User and Developer Experience
+=============================
+
+Bundle Size
+-----------
+
+Each MFE bundles all of its own dependencies, even if they're the same version as another MFE. This means that as a user navigates between MFEs, they end up downloading the same set of common dependencies over and over again. This can result in megabytes of overhead for the average user navigating between a few MFEs, and slows down the entire experience.
+
+Full Page Refreshes
+-------------------
+
+Each MFE has its own index.html page, and needs to load all its own dependencies from scratch whenever you navigate to it. This means that the browser performs a full page refresh each time a user transfers from MFE to MFE.
+
+Build time
+----------
+
+Each MFE must bundle all its own dependencies at build time, creating significant overhead for the Webpack build process. When building multiple MFEs, this additional, repeated overhead adds up quickly, making builds prohibitively slow.
+
+Dependency Maintenance
+----------------------
+
+Since each MFE has its own complete set of dependencies, the overhead of keeping them all up to date can be overwhelming. Security patches, bug fixes, new features, and breaking changes all add up and create a significant maintenance burden.
+
+Composability
+=============
+
+We have no clean, low-overhead way of composing components from multiple MFEs into a single page, or extending an MFE with additional functionality.
+
+The reality of MFEs is that while each application may represent a cohesive domain or bounded context, sometimes you need content and functionality from more than one domain on the page at the same time.
+
+Alternately, providers may want to show different *versions* of MFEs to different users while keeping the rest of the app (header, navigation, other MFEs) unchanged.
+
+We aren't without options here, but what we have today has its own issues. It's worth walking through what we have today.
```

Comment:
Again I'm wary of "today" - an OEP seeks to be a living document, driving our architecture for years to come. I would expect to, in 3 years, point a new dev to the frontend to this document to learn about composibility, and I find language like "today" or "currently" to work against that goal. I'm not sure how to suggest a rewording, what's your intention behind including it? How could we adapt that to the next few sections ("Current Option")?

Reply:
Gotcha... I'm going to do a general pass with an eye toward today-ness.  Rather than talking about "MFEs today", we should be talking about "Siloed MFEs" or some such, I think, talking about the architecture without talking about it being 'current'.